logisland-docs/developer.rst

```diff
@@ -2,23 +2,247 @@
 Developer Guide
 ===============
 
-Here is a set of information which you should read if you want to contribute to the project.
+This document summarizes information relevant to logisland committers and contributors. 
+It includes information about the development processes and policies as well as the tools we use to facilitate those.
 
 
+
+Workflows
+---------
+
+This section explains how to perform common activities such as reporting a bug or merging a pull request.
+
+
+Coding Guidelines
++++++++++++++++++
+
+Basic
+_____
+
+1. Avoid cryptic abbreviations. Single letter variable names are fine in very short methods with few variables, otherwise make them informative.
+2. Clear code is preferable to comments. When possible make your naming so good you don't need comments. When that isn't possible comments should be thought of as mandatory, write them to be read.
+3. Logging, configuration, and public APIs are our "UI". Make them pretty, consistent, and usable.
+4. Maximum line length is 130.
+5. Don't leave TODOs in the code or FIXMEs if you can help it. Don't leave println statements in the code. TODOs should be filed as github issues.
+6. User documentation should be considered a part of any user-facing the feature, just like unit tests. Example REST apis should've accompanying documentation.
+7. Tests should never rely on timing in order to pass.
+8. Every unit test should leave no side effects, i.e., any test dependencies should be set during setup and clean during tear down.
+
+Java
```

Comment:
Missing some information related to project preference between Java and Scala code.